webrender/src/scene.rs

```diff
@@ -99,7 +99,7 @@ pub struct ScenePipeline {
 pub struct Scene {
     pub root_pipeline_id: Option<PipelineId>,
     pub pipelines: FastHashMap<PipelineId, Arc<ScenePipeline>>,
-    pub pipeline_epochs: FastHashMap<PipelineId, Epoch>,
+    pub pipeline_epochs: FastHashMap<PipelineId, (Epoch, BuildState)>,
```

Comment:
hmm, this seems misplaced. Scene is the source for building, it's not supposed to know about build artifacts (like this flag)

Reply:
Ok. I should be able to take this out, will update the patch.

Reply:
@kvark Would the frame builder be a better place for this? The flattener can construct the map with all the built and unbuilt entries, pass it to the frame builder, and the render_backend can read it from the frame builder in `updated_pipeline_info` instead of the from the scene.

Reply:
sounds reasonable